### DISCUSSION TITLE
Request: Proxy support

### DISCUSSION THREAD

**Discussion by owendswang** [BlafKing/sd-civitai-browser-plus#218] [Ideas]
(Moving from CivitAI.)

I didn't dig into the interface codes. So I just tried to add proxies to http requests.

There are 2 kinds of http request as I see from the py scripts (civitai_api.py, civitai_file_manager.py).

1. 'reqeusts' package:
proxies = {'http': 'http://192.168.1.6:8118', 'https': 'http://192.168.1.6:8118'}
requests.get(url, proxies=proxies)
2. 'urllib' package:
proxies = {'http': 'http://192.168.1.6:8118', 'https': 'http://192.168.1.6:8118'}
proxy = urllib.request.ProxyHandler(proxies)
opener = urllib.request.build_opener(proxy)

For current version v3.4.1, those line numbers are as below.
'civitai_api.py': Line 635,648,1255
'civitai_file_manager.py': Line 233,194,657,904,972

Hope this would help you to add proxy support in the web interface settings panel.

**Comment by BlafKing** [comment]
The latest version now includes proxy support :)

Do note that it's required to use proxies that support HTTPS, so usually those are Socks4 and Socks5 proxies!

**Reply by owendswang** [reply]
Wow, amazing! Thank you!
